pandas/core/frame.py

```diff
@@ -6960,8 +6964,12 @@ def explode(self, column: Union[str, Tuple]) -> "DataFrame":
         # TODO: use overload to refine return type of reset_index
         assert df is not None  # needed for mypy
         result = df[column].explode()
-        result = df.drop([column], axis=1).join(result)
-        result.index = self.index.take(result.index)
+        if ignore_index:
+            result = df.drop([column], axis=1).join(result)
```

Comment:
you can leave this line before the if/else, since it is now identical in both cases

Reply:
Yes, done. Thank you